src/Integration/MefServices/IActiveSolutionBoundTracker.cs

```diff
@@ -18,5 +18,11 @@ public interface IActiveSolutionBoundTracker : IDisposable
         /// Returns whether the active solution is bound to a SonarQube project
         /// </summary>
         bool IsActiveSolutionBound { get; }
+
+        /// <summary>
+        /// Event to notify subscribers when the analyzers of a solution might have changed.
+        /// This occurs when a new solution is opened, or the SonarQube binding status of the solution changes.
+        /// </summary>
+        event EventHandler AnalyzersChanged;
```

Comment:
See my comment abound all the pieces needed to determine. Solution changes, binding changes and a service are all required since we need to support two extremes - TE never opened, TE was open before the first solution was loaded.